packages/http-client/src/http-client.ts

```diff
@@ -121,6 +123,61 @@ export function execute<ReturnT>(executeFn: ExecuteHttpRequestFn<ReturnT>) {
   };
 }
 
+function encodeRequestQueryParameters(
+  parameter: Record<string, any> | undefined
+): Record<string, any> | undefined {
+  if (parameter) {
+    return Object.fromEntries(
+      Object.entries(parameter).map(([key, value]) => [
+        encodeURIComponent(key),
+        encodeURIComponent(value)
+      ])
+    );
+  }
+}
+/**
+ * This method does nothing and is only there to indicated that the call was made by Odata or OpenApi client and encoding is already done on filter and key parameters.
+ * @param params - Parameters which are returned
+ * @returns The parameters as they are without encoding.
+ * @internal
+ */
+export const encodeTypedClientRequest: ParameterEncoder = (
+  params: Record<string, any>
+) => params;
+
+function getEncodedParameters(
+  parameters: OriginOptionsInternal,
+  requestConfig: HttpRequestConfigWithOrigin
+): OriginOptionsInternal {
+  const { parameterEncoder } = requestConfig;
+  // Custom Parameter encoder given->use it
+  if (
+    parameterEncoder &&
+    parameterEncoder.name !== encodeTypedClientRequest.name
+  ) {
+    return Object.fromEntries(
+      Object.entries(parameters).map(([key, value]) => [
+        key,
+        parameterEncoder(value)
+      ])
+    );
+  }
+
+  // If the parameterEncoder is undefined the executeHttpRequest() was used directly -> encode the request Parameters as wel
+  const requestParameterEncoder = !parameterEncoder
```

Comment:
When reaching this line, `parameterEncoder` should be `encodeTypedClientRequest`.
Then why should we check it in this line?